cvpy/biomedimage/BiomedImage.py

```diff
@@ -241,3 +242,62 @@ def rename_columns():
 
         # Change column names in the ImageTable
         rename_columns()
+
+    def morphological_gradient(self, images: ImageTable, kernel_width: int = 3,
+                               kernel_height: int = 3, copy_vars: List[str] = None) -> ImageTable:
+        """
+        Compute morphological gradient for given 3D grayscale image table.
+
+        Parameters
+        ------------
+        images : :class:'ImageTable'
+            Specifies the image table that contains 3D grayscale image.
+        kernel_width : :class:'int'
+            Specifies the kernel width.
+        kernel_height : :class:'int'
+            Specifies the kernel height.
+        copy_vars : :class:'List[str]'
+            Specifies which columns to copy to the output image table.
+
+        Returns
+        ------------
+        :class:'ImageTable'
+        """
+
+        conn = self._cas_session
+        random_name_generator = RandomNameGenerator()
+
+        # Export images from 3d to 2d
+        name_image_2d = random_name_generator.generate_name()
+        image_2d = conn.CASTable(name=name_image_2d, replace=True)
+        conn.biomedimage.processbiomedimages(images=dict(table=images.table),
+                                             steps=[dict(stepparameters=dict(steptype='export'))],
+                                             casout=image_2d,
+                                             copyvars=copy_vars)
+
+        # Compute morphological gradient of 2d images
+        name_morph_grad_2d = random_name_generator.generate_name()
```

Comment:
I know the chances of this are slim, but maybe we should check to make sure that the generated name isn't a CAS table that already exists in the user's caslib.

Reply:
Yes, I will doing that as part of a follow-up story - 4333.